web/packages/teleport/src/Notifications/Notification.tsx

```diff
@@ -78,11 +86,41 @@ export function Notification({
   const formattedDate = formatDate(notification.createdDate);
 
   function onMarkAsClicked() {
-    // TODO rudream - add mark as clicked functionality
+    const newNotificationState = clicked
+      ? NotificationState.UNSPECIFIED
+      : NotificationState.CLICKED;
+
+    const initialState = clicked;
+    // Update the visual state right away for a snappier UX.
+    setClicked(!clicked);
```

Comment:
Can the notification be "unclicked"?
Because `setClicked(true)` could be enough.

Also, we could do this in a different way, using `useAsync`:
```ts
  const [markAsClickedAttempt, markAsClicked] = useAsync(() =>
    ctx.notificationService.upsertNotificationState(clusterId, {
      notificationId: notification.id,
      notificationState: NotificationState.CLICKED,
    })
  );

// and then:

      <Container
        clicked={
          notification.clicked ||
          markAsClickedAttempt.status === 'processing' ||
          (markAsClickedAttempt.status === 'success' &&
            markAsClickedAttempt.data.notificationState ===
              NotificationState.CLICKED)
        }
        onClick={onNotificationClick}
        className="notification"
      >
```

Benefit of this is that we won't have to remember about calling `setClicked` when the response comes, it will also work properly if the `notification` passed through props becomes clicked.

We could do the same for hiding notifications.

Btw, don't we want to show errors to the user? It could be some small red text on the notification. This makes it easier to notice when something is not working properly.